src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,98 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] storage;
+    private int size;
+
+    public ArrayList() {
+        this.storage = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkArraySizeAndResize();
+        size++;
+        checkIndex(index);
```

Comment:
```suggestion
        checkIndex(index);
        checkArraySizeAndResize();
        size++;
```
In the case of an incorrect index, redundant actions will not be performed.

Reply:
варто залишити мій варіант 
інакше прийдеться переробити метод checkIndex